pkg/wireguard/agent/agent.go

```diff
@@ -517,7 +523,7 @@ func (a *Agent) deletePeerByPubKey(pubKey wgtypes.Key) error {
 }
 
 // updatePeerByConfig updates the WireGuard kernel peer config based on peerConfig p
-func (a *Agent) updatePeerByConfig(p *peerConfig) error {
+func (a *Agent) updatePeerByConfig(p *peerConfig, forceUpdate bool) error {
```

Comment:
how about renaming `forceUpdate` to `pubKeyChanged`, if that's the only case requiring force update when `len(peer.needsInsert) == 0`?

Reply:
actually, I still have some doubts here: do we really need `forceUpdate` in the case of pubkey change? We already have:
```go
	// Reinitialize peer if its public key changed.
	if peer != nil && peer.pubKey != pubKey {
		log.WithField(logfields.NodeName, nodeName).Debug("Pubkey has changed")
		if err := a.deletePeerByPubKey(peer.pubKey); err != nil {
			return err
		}
		peer = nil
	}
```
the peer is bound to be re-initialized with needsInsert entries? We don't need forceUpdate to handle key change situation?

Reply:
Thanks for the comment, this helped me realizing that this is not needed at all.
Back to when I traced the n. of calls to the `updatePeerByConfig` with respective parameters, I noticed there are cases with no IPs to insert/remove at all. Though, I was not thinking this might come from a special case in which we call `UpdatePeer` (and therefore `updatePeerByConfig`) with no changes at all, which is on (periodic) `NodeValidateImplementation`.
However, in that case the 1st call to `UpdatePeer` (or upon finishing restoration) would have already configured the device, therefore there's no need to force it again.
 I am going to fix this commit removing such additional parameter as not needed.